### PR TITLE
327049 - Credential Validation.

### PR DESCRIPTION
**Pull request by jakebrown58**
Better UI expereince for credential validation.
* If setMasterCredentials fails, it'll now trap that in an error block
that will show up in the front end in red text.

### CODE REVIEW

**Comment by jakebrown58** [comment]
I need to add tests.

File: test/server/tasks/3-validate-machine.js [hunk 1]
```diff
@@ -90,6 +98,7 @@ describe('3-validate-machine', function() {
       stubSetConfigs      = sinon.stub(RingtailClient.prototype, 'setConfigs');
```
Comment (danjea) [inline test/server/tasks/3-validate-machine.js]: Consider loading all the stubs into an array so you can just call stubs.forEach((stub) => {stub.restore();}) or something similar.

File: test/server/tasks/3-validate-machine.js [hunk 2]
```diff
@@ -181,6 +191,17 @@ describe('3-validate-machine', function() {
         .catch(done);
     });
 
+    it('sets the master runner credentials', function(done) {
```
Comment (danjea) [inline test/server/tasks/3-validate-machine.js]: Should there be a test for the new exception handling path you added to the task runner?

File: test/server/tasks/3-validate-machine.js (hunk 1 above)
Comment (jakebrown58) [reply test/server/tasks/3-validate-machine.js]: cool - next time I'm in there, I'll give that a look.   not a bad idea.

also would like to refactor this test class for async / await style.

File: test/server/tasks/3-validate-machine.js (hunk 2 above)
Comment (jakebrown58) [reply test/server/tasks/3-validate-machine.js]: yes, probably.   this is really just a smoke test.  i can add a task for refactoring the tests to make them more meaningful.

**Pull request merged**